state-network.md

```diff
@@ -255,53 +269,297 @@ sedes      := Container(enrs: List[byte_list, max_length=32], payload: byte_list
+    - DHT nodes receiving trie nodes extract the trie nodes from the anchor proof to perform *recursive gossip*.
+    - DHT nodes receiving "leaf" nodes initiate gossip of the leaf proofs (for stage 3)
+- Stage 3:
+    - DHT nodes receiving leaf proofs perform *neighborhood gossip* to spread the data to nearby interested DHT nodes.
+
 
 ```
-message_frame := length_prefix | message
-length_prefix := bytes4
-message       := encode(proof)
-proof         := TBD
+    +-------------------------+
+    | Stage 1: data ingress   |
+    +-------------------------+
+    |                         |
+    | Bridge node initializes |
+    | trie node gossip        |
+    |                         |
+    +-------------------------+
+            |
+            v
+    +---------------------------+
+    | Stage 2: trie nodes       |
+    +---------------------------+
+    |                           |
+    | A: neighborhood gossip of |
+    |    trie node and proof    |
+    |                           |
+    | B: initialization of      |
+    |    gossip for proof trie  |
+    |    nodes.                 |
+    |                           |
+    | C: initialization of      |
+    |    leaf proof gossip      |
+    |                           |
+    +---------------------------+
+            |
+            v
+    +----------------------------+
+    | Stage 3: leaf proofs       |
+    +----------------------------+
+    |                            |
+    | neighborhood gossip of     |
+    | leaf proofs                |
+    |                            |
+    +----------------------------+
+
 ```
 
-> The exact semantics of how we encode proofs are under active development and subject to major changes.
 
-We encode proofs as a stream of `(path, node)` pairs which have been sorted in preorder traversal ordering.  We can save some bytes by de-duplicating the `path` components since each path will tend to have a common prefix with the previous path.
+In this context the term "neighborhood gossip" refers to a DHT node
+re-distributed data that they were interested in to the DHT nodes nearby who
+are also interested.
+
+The phrase "initialization of XXX gossip" refers to finding the DHT nodes that
+are responsible for XXX and offering the data to them.
+
+
+#### Stage 1: Data Ingress
+
+
+The first stage of gossip is performed by a bridge node. Each time a new block
+is added to their view of the chain, a set of merkle proofs which are all
+anchored to `Header.state_root` is generated which contains.
+
+- Account trie Data:
+    - All of the intermediate and leaf trie nodes from the account trie necessary to prove new and modified accounts.
+    - All of the intermediate and leaf trie nodes from the account trie necessary for exclusion proofs for deleted accounts.
+- Contract Storage trie data:
+    - All of the intermediate and leaf trie nodes from each contract storage trie necessary to prove new and modified storage slots.
```

Comment:
We might need to do more thinking on the proofs of new storage data. A node doesn't know that it's responsible for storage data until it can prove that the storage data is part of an account. That means (at least temporarily) proving the account with the storage root. I believe that we've been assuming storage proof gossip does *not* gossip the account proof, but for a node that doesn't know it's responsible for the slot yet, that makes it impossible to immediately prove inclusion for its local storage.